subt/cloudsim2osgar.py

```diff
@@ -107,8 +114,9 @@ def __init__(self, robot_name):
             setattr(self, handler.__name__+"_count", 0)
             rospy.Subscriber(name, type, handler)
 
-        rospy.sleep(2)
-        rospy.spin()
+        while rospy.ok():
+            channel, bytes_data = self.bus.listen()
```

Comment:
is this blocking? Termination is also Shutdown exception?

Reply:
yes, this is blocking - termination is SIGINT (raises KeyboardException) - we will see how this works out in the wild (aka _needs testing_)

Reply:
not working, hmm

Reply:
meaning it stays running after everything quits and it needs to be killed with -9

Reply:
fixed in 44fe5e3